### PR TITLE
SASS-8594 - Rentals & RaR - About section: Implement DES/IF Create & …

### PR DESCRIPTION
**Pull request by modivilioglu**
…Update Annual property submission API

### CODE REVIEW

File: app/controllers/JourneyAnswersController.scala [hunk 1]
```diff
@@ -133,6 +133,25 @@ class JourneyAnswersController @Inject() (
...
+    auth.async { implicit request =>
+      withJourneyContextAndEntity[RaRAbout](
+        taxYear,
+        incomeSourceId,
+        nino,
+        JourneyName.PropertyRentalsAndRentARoomAbout,
+        request
+      ) { (ctx, rarAbout) =>
+        handleResponse(CREATED) {
+          propertyService.saveRaRAbout(ctx, nino, rarAbout)
```
Comment (arunkumar461) [inline app/controllers/JourneyAnswersController.scala:150]: `RaRAbout` Are we not creating separate case class? we need to capture more fields

**Review by arunkumar461 — CHANGES_REQUESTED** [review]

File: app/controllers/JourneyAnswersController.scala (hunk 1 above)
Comment (modivilioglu) [reply app/controllers/JourneyAnswersController.scala]: Done. Thanks

**Review by vivekhmrc — DISMISSED** [review]

File: app/models/RentalsAndRaRAbout.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+package models
+
+import models.request.{BalancingCharge, ClaimExpensesOrRRR}
+import play.api.libs.json.Json
+
+final case class RentalsAndRaRAbout(
+  ukRentARoomJointlyLet: Boolean,
+  totalIncomeAmount: BigDecimal,
+  balancingCharge: BalancingCharge,
```
Comment (arunkumar461) [inline app/models/RentalsAndRaRAbout.scala:25]: balancingCharge in about section?

File: app/models/responses/PropertyAnnualSubmission.scala [hunk 3]
```diff
@@ -121,6 +122,45 @@ object PropertyAnnualSubmission {
     resultWithSbas
   }
 
+  def fromRentalsAndRentARoomAbout(
```
Comment (arunkumar461) [inline app/models/responses/PropertyAnnualSubmission.scala:125]: Let's keep an eye on this file, as it’s growing. We also need to consider how much of it we can reuse.

File: app/models/request/sba/SbaInfo.scala [hunk 4]
```diff
@@ -34,7 +34,7 @@ object StructureBuildingFormGroup {
 
 final case class SbaInfo(
   claimStructureBuildingAllowance: ClaimStructureBuildingAllowance,
-  structureBuildingFormGroup: List[StructureBuildingFormGroup]
+  sbas: List[StructureBuildingFormGroup]
```
Comment (arunkumar461) [inline app/models/request/sba/SbaInfo.scala:37]: Rename case class also?

**Review by arunkumar461 — CHANGES_REQUESTED** [review]

File: app/models/RentalsAndRaRAbout.scala (hunk 2 above)
Comment (arunkumar461) [reply app/models/RentalsAndRaRAbout.scala:25]: image

File: app/models/request/sba/SbaInfo.scala (hunk 4 above)
Comment (modivilioglu) [reply app/models/request/sba/SbaInfo.scala:37]: Done.

File: app/models/responses/PropertyAnnualSubmission.scala (hunk 3 above)
Comment (modivilioglu) [reply app/models/responses/PropertyAnnualSubmission.scala:125]: yes.

File: app/models/RentalsAndRaRAbout.scala (hunk 2 above)
Comment (modivilioglu) [reply app/models/RentalsAndRaRAbout.scala:25]: From the prototype, there is balancing charge as far as I can see.

**Review by arunkumar461 — APPROVED** [review]

**Pull request merged**